.development/generate_menu_screenshots.py

```
@@ -0,0 +1,323 @@
+    Returns a set of class names referenced in assignments.
+    """
+    main_py = os.path.join(os.path.dirname(ROOT), "main.py")
+    if not os.path.exists(main_py):
+        return set()
+    names = set()
+    try:
+        with open(main_py, "r", encoding="utf-8") as fh:
+            for line in fh:
+                line = line.strip()
+                # simple heuristic: look for `name = ClassName(`
+                if "=" in line and "(" in line and line.endswith(")") or "(" in line:
```

Ruff (0.14.13)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)